presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ApplyNode.java

```diff
@@ -37,22 +39,30 @@
      */
     private final List<Symbol> correlation;
 
+    /**
+     * Non-scalar predicates which operate on subqueries (e.g: "expr IN (SELECT 1)")
+     */
+    private final Map<Symbol, Expression> subqueryPredicates;
```

Comment:
why do you limit that only to non scalar? I would store all the subquery expressions